modules/azure/disk.go

```diff
@@ -0,0 +1,103 @@
+package azure
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-07-01/compute"
+	"github.com/gruntwork-io/terratest/modules/testing"
+	"github.com/stretchr/testify/require"
+)
+
+// DiskExists indicates whether the specified Azure Managed Disk exists
+// This function would fail the test if there is an error.
+func DiskExists(t testing.TestingT, diskName string, resGroupName string, subscriptionID string) bool {
+	exists, err := DiskExistsE(t, diskName, resGroupName, subscriptionID)
+	require.NoError(t, err)
+	return exists
+}
+
+// DiskExistsE indicates whether the specified Azure Managed Disk exists
+func DiskExistsE(t testing.TestingT, diskName string, resGroupName string, subscriptionID string) (bool, error) {
+	// Get the Disk object
+	_, err := GetDiskE(t, diskName, resGroupName, subscriptionID)
+	if err != nil {
+		if ResourceNotFoundErrorExists(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+// GetDiskType returns the Disk Storage Account Type of the Azure Managed Disk
+// This property also accessible from the VM client disk storage object but only works
+// when the VM is online, while this direct call to GetDiskType always works.
+// This function would fail the test if there is an error.
+func GetDiskType(t testing.TestingT, diskName string, resGroupName string, subscriptionID string) compute.DiskStorageAccountTypes {
+	diskType, err := GetDiskTypeE(t, diskName, resGroupName, subscriptionID)
+	require.NoError(t, err)
+	return diskType
+}
+
+// GetDiskTypeE returns the Disk Storage Account Type of the Azure Managed Disk
```

Comment:
NIT: The extra comments in `GetDiskType` applies here as well, so it should be replicated.

Reply:
Updated